client/src/views/page-components/MainPage/MainPage.js

```diff
@@ -1,23 +1,112 @@
-import UserSlide from "./UserSlide/UserSlide.js";
+import { SAMPLE_PRODUCTS_STATE } from '../../../dummy-file.js';
+import { createDOMwithSelector } from '../../../utils/createDOMwithSelector.js';
+import CategorySlide from './CategorySlide/CategorySlide.js';
+import CreateProductButton from './CreateProductButton/CreateProductButton.js';
+import CreateProductSlide from './CreateProductSlide/CreateProductSlide.js';
+import MainPageNavBar from './MainPageNavBar/MainPageNavBar.js';
+import MenuSlide from './MenuSlide/MenuSlide.js';
+import ProductsWrap from './ProductsWrap/ProductsWrap.js';
+import TownModal from './TownModal/TownModal.js';
+import TownSlide from './TownSlide/TownSlide.js';
+import UserSlide from './UserSlide/UserSlide.js';
 
-function MainPage(params) {
+function MainPage() {
+  /**
+   * State
+   * currentlyopendSlide
+   * - 현재 오픈되어 있는 슬라이드 입니다.
+   * - 들어올 수 있는 값은 null | category | user | town | create | menu 입니다.
+   */
   this.state = {
-    name: "종호",
+    products: SAMPLE_PRODUCTS_STATE,
+    currentlyOpenedSlide: null,
+    isTownModalOpened: false,
   };
-  this.$main = document.querySelector("#mainPage");
-
-  const userSlide = new UserSlide({
-    $main: this.$main,
-    onHandler: () => {
-      this.setState({ ...this.state, name: "기석" });
-    },
-    initialState: this.state.name,
-  });
 
   this.setState = (nextState) => {
     this.state = nextState;
-    userSlide.setState({ name: this.state.name });
+
+    triggerSlides(this.state.currentlyOpenedSlide);
+    triggerTownModal(this.state.isTownModalOpened);
```

Comment:
triggerSlides, triggerTownModal의 경우는 state가 변경되는 것 보다는
화면이 변하는 느낌이 강한 것 같아서 render 함수 안에서 실행하면 어떨까요??

```suggestion
this.setState = (nextState) => {
   this.state = nextState;
   this.render();
 }

this.render = () => {
    triggerSlides(this.state.currentlyOpenedSlide);
    triggerTownModal(this.state.isTownModalOpened);
  }
```

Reply:
저한테도 말씀해주신게 의미론적으로 더 와닿네요!! 수정하겠습니다 :)